packages/neon/neon/lib/src/bloc/result.dart

```diff
@@ -67,12 +123,29 @@ class Result<T> {
         isCached: isCached ?? this.isCached,
       );
 
+  /// Returns whether this snapshot contains a non-null [error] value.
+  ///
+  /// This is always true if the asynchronous computation's last result was
+  /// failure.
+  /// A result may both have an error and data.
   bool get hasError => error != null;
 
+  /// Returns whether this snapshot contains a non-null [data] value.
+  ///
+  /// This can be false even when the asynchronous computation has completed
+  /// successfully, if the computation did not return a non-null value. For
+  /// example, a [Future<void>] will complete with the null value even if it
+  /// completes successfully.
```

Comment:
I feel like this is rather obvious and implied. If T is a nullable type, then this is totally expected.

Reply:
I agree that this might be expected but I don't think it's obvious without looking at the code.

If we publish neon on pub.dev the documentation will be hosted with it. Someone browsing the api reference without looking at the code might find this information helpful.